src/SpacetimeDBClient.cs

```diff
@@ -141,8 +142,22 @@ struct DbOp
         [Obsolete]
         public event Action<Exception>? onSendError;
 
+        /// <summary>
+        /// Dictionary of legacy subscriptions, keyed by request ID rather than query ID.
+        /// Only used for `SubscribeToAllTables()`.
+        /// </summary>
+        private readonly Dictionary<uint, ISubscriptionHandle> legacySubscriptions = new();
```

Comment:
Hm, I assume you made id increment atomic to allow `.Subscribe()` from multiple threads.

I guess that's not the usecase we thought of in the beginning, but if we want to allow that, should those dictionaries also be made `ConcurrentDictionary` instead?

Reply:
Alternatively, I think it's also fine to revert to regular `int` and leave support for `.Subscribe` from multiple threads for a future PR. It's certainly not very common usecase.

Reply:
I was just blindly copying @jsdt's implementation. Happy to simplify.